build-info-extractor-maven3/src/main/java/org/jfrog/build/extractor/maven/BuildDeploymentHelper.java

```diff
@@ -105,13 +100,27 @@ public void deploy( Build                          build,
         }
 
         if (isDeployArtifacts(clientConf, wereThereTestFailures, deployableArtifacts)) {
+            logExcludedArtifacts(build);
             deployArtifacts(clientConf, deployableArtifacts);
         }
         if (isPublishBuildInfo(clientConf, wereThereTestFailures)) {
             publishBuildInfo(clientConf, build, buildInfoAggregated);
         }
     }
 
+
+    private void logExcludedArtifacts(Build build) {
```

Comment:
We can probably remove this method and add the logging in the **BuildInfoRecorder.addArtifactsToCurrentModule** method. Since this PR removes the usage of the *pathConflicts* in the *BuildDeploymentHelper.deployArtifacts* method, the code in *addArtifactsToCurrentModule* should probably be changed from:
```
            if (artifactFile != null && artifactFile.isFile()) {
                // If excludeArtifactsFromBuild and the PatternMatcher found conflict, add the excluded artifact to the excluded artifact set.
                if (excludeArtifactsFromBuild && PatternMatcher.pathConflicts(deploymentPath, patterns)) {
                    module.addExcludedArtifact(artifact);
                } else {
                    module.addArtifact(artifact);
                }
                addDeployableArtifact(artifact, artifactFile, moduleArtifact.getGroupId(), artifactId, artifactVersion, artifactClassifier, artifactExtension);
            }
```
to something like:
```
            if (artifactFile != null && artifactFile.isFile()) {
                // If excludeArtifactsFromBuild and the PatternMatcher found conflict, add the excluded artifact to the excluded artifact set.
                boolean pathConflicts = PatternMatcher.pathConflicts(deploymentPath, patterns)
                if (excludeArtifactsFromBuild && pathConflicts) {
                    module.addExcludedArtifact(artifact);
                } else {
                    module.addArtifact(artifact);
                }
                if (!pathConflicts) {
                    addDeployableArtifact(artifact, artifactFile, moduleArtifact.getGroupId(), artifactId, artifactVersion, artifactClassifier, artifactExtension);
                } else {
                      <Logging should be added here>
                }
            } 
```
I suggest we test this though with Maven Project, Free Style and Pipeline.